### PR TITLE
Added CI/CD/CT

### PR DESCRIPTION
**Pull request by ilkerhalil**
fixes #<issue number> <issue synopsis>

<Comments describing your change. Not all changes need this.>

Note that the above format should be used in your git commit comments.
You agree that by submitting a PR, you have read and agreed to our
contributing guidelines.

### CODE REVIEW

**Review by TYoungSL — APPROVED** [review]
LGTM

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Dependencies for linux
+        run: |
+          sudo apt-get install -y ninja-build
+      - name: Download mbedtls Linux Release
+        run: |
+          wget https://github.com/ilkerhalil/mbedtls/releases/download/v3.2.0/mbedtls-linux.tar.gz
+          wget https://github.com/ilkerhalil/mbedtls/releases/download/v3.2.0/mbedtls-headers.tar.gz
```
Comment (Rjvs) [inline .github/workflows/cd.yml:18]: I'll fork mbedtls so we can get the files from an SL repo

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      - name: Build nng
+        run: |
+          mkdir build
+          cd build
+          cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DNNG_ELIDE_DEPRECATED=ON -DNNG_TOOLS=OFF  ..
+          cmake --build . --config Release
+      # - name: Test nng
+      #   run: |
+      #     ctest -C Release --output-on-failure
+      #  working-directory: build
```
Comment (Rjvs) [inline .github/workflows/cd.yml:218]: These should be turned back on.

**Review by Rjvs — APPROVED** [review]
We'll need to fix a few issues but LGTM

**Pull request merged**